### PR TITLE
Cancel functionality added for Nodes and Edge dialogs. Remembering re…

### PR DESCRIPTION
**Pull request by ajedru**
…cent file for further saving added

### CODE REVIEW

File: violet-framework/src/main/java/com/horstmann/violet/framework/dialog/IRevertableProperties.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    /**
+     * Signalize implementing class that it should cache it's cache-able data at time of method call
+     */
+    void beforeUpdate();
+
+    /**
+     * Signalize implementing class that it should revert all changes to it's cache-able properties and assign them their
+     * old cached values
+     */
+    //TODO: rename to something with more general meaning
```
Comment (NdV66) [inline violet-framework/src/main/java/com/horstmann/violet/framework/dialog/IRevertableProperties.java]: TODO in production code? Please explain.

**Review by NdV66 — CHANGES_REQUESTED** [review]

File: violetplugin-communicationdiagram/src/main/java/com/horstmann/violet/product/diagram/communication/node/ActorNode.java [hunk 2]
```diff
@@ -172,6 +176,20 @@ public void setName(SingleLineText newValue)
 //        getContent().refresh();
```
Comment (NdV66) [inline violetplugin-communicationdiagram/src/main/java/com/horstmann/violet/product/diagram/communication/node/ActorNode.java]: why comment

File: violetplugin-statediagram/src/main/java/com/horstmann/violet/product/diagram/state/edge/StateTransitionEdge.java [hunk 3]
```diff
@@ -136,8 +135,14 @@ private Rectangle2D getLabelBounds()
         // need a dummy image to get a Graphics to
```
Comment (NdV66) [inline violetplugin-statediagram/src/main/java/com/horstmann/violet/product/diagram/state/edge/StateTransitionEdge.java]: why comment

**Review by NdV66 — CHANGES_REQUESTED** [review]

File: violet-framework/src/main/java/com/horstmann/violet/framework/dialog/IRevertableProperties.java (hunk 1 above)
Comment (ajedru) [reply violet-framework/src/main/java/com/horstmann/violet/framework/dialog/IRevertableProperties.java]: That was change request that didnt make it. Forgot to remove todo, will fix it.

File: violetplugin-communicationdiagram/src/main/java/com/horstmann/violet/product/diagram/communication/node/ActorNode.java (hunk 2 above)
Comment (ajedru) [reply violetplugin-communicationdiagram/src/main/java/com/horstmann/violet/product/diagram/communication/node/ActorNode.java]: This and other below are some comments I dont recall making - must have been cloned from the other repo, will fix as well.

File: violet-framework/src/main/java/com/horstmann/violet/framework/dialog/IRevertableProperties.java (hunk 1 above)
Comment (NdV66) [reply violet-framework/src/main/java/com/horstmann/violet/framework/dialog/IRevertableProperties.java]: So, fix it. We're waiting.

Comment (ajedru) [reply violet-framework/src/main/java/com/horstmann/violet/framework/dialog/IRevertableProperties.java]: So wait a little longer, long way before I make it home back from work.

**Review by orchowski — APPROVED** [review]

**Review by abizukojc — APPROVED** [review]

**Pull request merged**